File: examples/web/src/news/newsFormat.ts
export const SNIPPET_MAX = 160;

import { isStubBody } from "./fullText";

export function plainSnippet(raw: string | undefined, maxChars = SNIPPET_MAX): string {
  const text = (raw ?? "")
    .replace(/<[^>]+>/g, " ")
    .replace(/\s+/g, " ")
    .trim();
  if (text.length <= maxChars) return text;
  return `${text.slice(0, maxChars).trimEnd()}…`;
}

export function listSnippet(raw: string | undefined, maxChars = SNIPPET_MAX): string {
  if (isStubBody(raw)) return "";
  const text = plainSnippet(raw, maxChars);
  return /https?:\/\//.test(text) ? "" : text;
}

export function ageLabel(publishedAt: number | undefined, now = Date.now()): string {
  if (publishedAt == null) return "";
  const minutes = Math.max(0, now - publishedAt) / 60_000;
  if (minutes < 60) return "now";
  const hours = Math.floor(minutes / 60);
  if (hours < 48) return `${hours}h`;
  return `${Math.floor(hours / 24)}d`;
}

File: examples/android/app/src/main/java/org/aetherfeed/app/news/NewsFormat.kt
package org.aetherfeed.app.news

const val SNIPPET_MAX = 160

enum class AgeUnit { Empty, Now, Hours, Days }

data class AgeLabel(val unit: AgeUnit, val count: Int = 0)

fun plainSnippet(raw: String?, maxChars: Int = SNIPPET_MAX): String {
    val text = raw.orEmpty()
        .replace(Regex("<[^>]+>"), " ")
        .replace(Regex("\\s+"), " ")
        .trim()
    if (text.length <= maxChars) return text
    return text.take(maxChars).trimEnd() + "…"
}

fun listSnippet(raw: String?, maxChars: Int = SNIPPET_MAX): String {
    if (isStubBody(raw)) return ""
    val text = plainSnippet(raw, maxChars)
    return if (Regex("https?://").containsMatchIn(text)) "" else text
}

fun ageLabel(publishedAt: Long?, now: Long = System.currentTimeMillis()): AgeLabel {
    if (publishedAt == null) return AgeLabel(AgeUnit.Empty)
    val minutes = ((now - publishedAt).coerceAtLeast(0L)) / 60_000L
    if (minutes < 60) return AgeLabel(AgeUnit.Now)
    val hours = (minutes / 60).toInt()
    if (hours < 48) return AgeLabel(AgeUnit.Hours, hours)
    return AgeLabel(AgeUnit.Days, hours / 24)
}
